client/state/preferences/actions.js

```diff
@@ -0,0 +1,52 @@
+/**
+ * Internal dependencies
+ */
+import wpcom from 'lib/wp';
+import {
+	PREFERENCES_SET,
+	PREFERENCES_RECEIVE,
+	PREFERENCES_REMOVE,
+	PREFERENCES_FETCH
+} from 'state/action-types';
+
+export const USER_SETTING_KEY = 'calypso_preferences';
+
+export function fetchPreferences() {
+	return ( dispatch ) => {
+		dispatch( { type: PREFERENCES_FETCH } );
+		return wpcom.me().settings().get().then( ( data ) => {
+			dispatch( {
+				type: PREFERENCES_RECEIVE,
+				data
+			} );
+		} );
+	};
+}
+
+const save = () => ( dispatch, getState ) => {
+	const settings = {};
+	settings[ USER_SETTING_KEY ] = getState().preferences.values;
+	wpcom.me().settings().update( JSON.stringify( settings ) ).then( ( data ) => {
+		dispatch( {
+			type: PREFERENCES_RECEIVE,
+			data
```

Comment:
Well, that not exactly how it works :)

We have `/me/settings` endpoint which behaves as you describe, **but** `calypso_settings` is one key -value pair in this endpoint and it is a serialized json.
So if we want to save data for calypso preferences, we need to pass whole updated settings object.
And if we update it, we get whole updated settings object in return.

So no, it will not wipe out anything, don't worry!

And tests will be there, but we're still not agreeing on naming and stuff, so I dont want to write them just so I need to update them multiple times.